File: solutions/cpp/kindergarten-garden/1/kindergarten_garden.cpp
#include "kindergarten_garden.h"

#include <array>
#include <string>

using namespace std;

namespace kindergarten_garden {

    Plants letterToPlant (char lt) {
         switch(lt) {
            case 'R' : return Plants::radishes;
            case 'C' : return Plants::clover;
            case 'G' : return Plants::grass;
            default: return Plants::violets;        
        }
    } 
  
    array<Plants,4> plants(string garden, string student){
        int studentNo = student.at(0) - 'A';
        int offset = garden.length()/2 + 1 + 2*studentNo;
        array<Plants,4> result =         
            {letterToPlant(garden.at(2*studentNo)),
            letterToPlant(garden.at(2*studentNo + 1)), 
            letterToPlant(garden.at(offset)),
            letterToPlant(garden.at(offset +1))};
        return result;
    }
    
}  // namespace kindergarten_garden

File: solutions/python/kindergarten-garden/1/kindergarten_garden.py
class Garden:
    def __init__(self, diagram, students=None):
        self.garden = diagram
        self.students = []
        if students == None:
            self.students = ["Alice", "Bob", "Charlie", "David", "Eve", "Fred", "Ginny", "Harriet", "Ileana", "Joseph", "Kincaid", "Larry"]
        else:
            self.students = students
            self.students.sort()        
        

    def letter_to_plant(self, letter):
        if letter == 'R':
            return "Radishes"
        elif letter == 'C':
            return "Clover"
        elif letter == 'V':
            return "Violets"
        else:
            return "Grass"

    def plants(self, student):
        if student not in self.students:
            raise ValueError("Invalid student")
        idx = self.students.index(student)
        offset = len(self.garden)//2 + 1 + 2*idx
        return [self.letter_to_plant(self.garden[2*idx]),
                 self.letter_to_plant(self.garden[2*idx+1]),
                 self.letter_to_plant(self.garden[offset]),
                 self.letter_to_plant(self.garden[offset+1])]